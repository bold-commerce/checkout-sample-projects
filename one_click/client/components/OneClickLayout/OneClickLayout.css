* {
  margin: 0;
  padding: 0;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

*::placeholder {
  color: rgba(0, 0, 0, 0.3);
}

*.stx-button:focus,
*.stx-input--has-focus,
*.stx-select--has-focus{
  box-shadow: none;
  outline: #2175c2 auto 2px;
}

*:focus-visible {
  outline: #2175c2 auto 2px;
  outline-offset: 4px;
}

.stx-checkbox {
  width: 16px;
  height: 16px;
}

html {
  box-sizing: border-box;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-smooth: always;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin: 0;
}

#checkout-app {
  display: grid;
  justify-items: center;
  width: 100vw;
  height: 100vh;
}

.Checkout__Main {
  max-width: 480px;
  width: 100%;
  position: relative;
  overflow-x: hidden;
  overflow-y: clip;
}

.Checkout__Form {
  position: relative;
  height: 100%;
  transition: all 0.6s;
  overflow-x: hidden;
}

.FieldSet {
  margin: 16px 0;
  padding: 0;
}

.FieldSet__Header {
  margin-bottom: 8px;
}

.FieldSet__Heading {
  font-size: 14px;
  line-height: 20px;
  font-weight: bold;
  color: rgba(0, 0, 0, 0.9);
}

.FieldGroup {
  display: grid;
  row-gap: 8px;
}

.Field {
  flex: 1;
}

.SelectField {
  display: block;
  cursor: pointer;
}

.SummaryBlock {
  position: relative;
  padding: 20px 0;
  border-bottom: 1px solid;
  border-color: rgba(0, 0, 0, 0.4);
}

body {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-smooth: always;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin: 0;
}

h1.Section__Title {
  padding-bottom: 16px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 36px;
  color: rgba(0, 0, 0, 0.9);
}

.Billing,
.Shipping,
.Summary {
  width: 100%;
}

.Sidebar {
  transition: all 0.6s;
  padding: 16px 32px;
}

.IndexGuest .Header,
.Billing .Header {
  border: none;
}

.IndexGuest a {
  color: #0a75c2;
  text-decoration: none;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}


.IndexGuest .stx-select__select-element,
.IndexGuest .stx-input {
  border-radius: 4px;
}

.IndexGuest a:visited {
  color: #0a75c2;
}

.IndexGuest__Customer,
.IndexGuest__ShippingAddress,
.IndexGuest__ShippingMethod,
.IndexGuest__Payment,
.IndexGuest__Footer {
  padding: 16px 0;
}

.IndexGuest__Summary {
  margin: 0 -32px;
  padding: 16px 0;
  background-color: #fafafa;
  border-width: 1px 0;
  border-style: solid;
  border-color:rgba(0, 0, 0, 0.4);
}

.IndexGuest__Title,
.IndexGuest__Summary__Btn h2,
.IndexGuest__Summary .stx-price__amount,
.IndexGuest__ShippingMethod .FieldSet__Heading {
  font-size: 18px;
  line-height: 24px;
  color: rgba(0, 0, 0, 0.9);
  font-weight: bold;
}

.IndexGuest__Summary__Btn {
  display: flex;
  border: none;
  background-color: unset;
  width: 100%;
  align-items: center;
  cursor: pointer;
}

.IndexGuest__Summary__Btn svg {
  overflow: initial;
  margin: 0 8px;
  width: 16px;
  height: 16px;
}

.IndexGuest__Summary__Btn h2 {
  width: 100%;
  text-align: left;
}

.IndexGuest__Summary__Btn .stx-price {
  margin-right: 32px;
}

.IndexGuest__Title,
.IndexGuest__ShippingMethod .FieldSet__Header {
  margin-bottom: 14px;
}

.IndexGuest__Customer {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}

.IndexGuest__Customer .stx-field--with-input {
  margin: 8px 0;
}

.IndexGuest__Customer .stx-input {
  margin: 0;
  height: 2.25rem;
}

.IndexGuest__Customer .stx-checkbox-field__checkbox {
  height: 16px;
  width: 16px;
}
.IndexGuest__Customer .stx-checkbox-field__field {
  padding-left: 26px;
}

.IndexGuest__Billing__Btn {
  width: 100%;
  padding: 12px 0;
  margin: 0;
  border: none;
  border-radius: 4px;
  background-color: #0a75c2;
}

.IndexGuest__Billing__Btn:disabled,
.Billing-CheckoutButton:disabled {
  background-color: rgba(0, 0, 0, 0.4);
}

.IndexGuest__Footer {
  display: grid;
  row-gap: 32px;
}

.IndexGuest__Rights {
  border-top: 1px solid #d9d9d9;
  padding-top: 32px;
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  color: rgba(0, 0, 0, 0.4);
}

.IndexGuest-footer .CheckoutButton__Container { 
  padding: 0;
}

@media only screen and (max-width: 768px) {
  .Sidebar {
    height: 100%;
    overflow-y: auto;
  }

  .Billing,
  .Shipping,
  .Summary {
    position: absolute;
    left: 0;
    top: 0;
  }

  .Sidebar--Show {
    transform: translate(0, 0);
    visibility: visible;
  }

  .Sidebar--Hide {
    transform: translate(100%, 0);
    visibility: hidden;
  }

  .IndexPage--Hide {
    transform: translate(-100%, 0);
    visibility: hidden;
  }
}

@media only screen and (min-width: 769px) {
  .Checkout__Main {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .Checkout__Form {
    transition: all 0.6s;
    overflow-y: hidden;
    max-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  .IndexPage--Hide {
    opacity: 0;
    display: none;
  }

  .IndexPage--Show {
    opacity: 1;
    display: block;
  }

  .Sidebar--Hide {
    opacity: 0;
    visibility: hidden;
    display: none;
  }

  .Sidebar--Show {
    opacity: 1;
    visibility: visible;
    display: block;
  }
}