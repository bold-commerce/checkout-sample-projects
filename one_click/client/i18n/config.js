import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;