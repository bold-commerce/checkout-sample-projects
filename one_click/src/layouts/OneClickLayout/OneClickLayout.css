* {
  margin: 0;
  padding: 0;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

*::placeholder {
  color: rgba(0, 0, 0, 0.3);
}

*.stx-button:focus,
*.stx-input--has-focus,
*.stx-select--has-focus{
  box-shadow: none;
  outline: #2175c2 auto 2px;
}

*:focus-visible {
  outline: #2175c2 auto 2px;
  outline-offset: 4px;
}

.stx-checkbox {
  width: 16px;
  height: 16px;
}

.stx-message {
  border-radius: 5px;
}

html {
  box-sizing: border-box;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-smooth: always;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin: 0;
}

#checkout-app {
  display: grid;
  justify-items: center;
}

.Checkout__Main {
  max-width: 480px;
  max-height:100%;
  width: 100%;
  height: auto;
  position: relative;
}

.Checkout__Form {
  position: relative;
  transition: all 0.6s;
  overflow-x: hidden;
  background: #fff;
  border-radius: 8px;
  box-shadow: 0px 1px 8px 2px #0000001A;
}

.Checkout__Form h2 {
  text-transform: none;
  letter-spacing: normal;
}

.FieldSet {
  margin: 16px 0;
  padding: 0;
}

.FieldSet__Header {
  margin-bottom: 8px;
}

.FieldSet__Heading {
  font-size: 14px;
  line-height: 20px;
  font-weight: bold;
  color: rgba(0, 0, 0, 0.9);
}

.FieldGroup {
  display: grid;
  row-gap: 8px;
}

.Field {
  flex: 1;
}

.SelectField {
  display: block;
  cursor: pointer;
}

.SummaryBlock {
  position: relative;
  padding: 20px 0;
  border-bottom: 1px solid;
  border-color: #D9D9D9;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  font-smooth: always;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin: 0;
}

h1.Section__Title {
  padding-bottom: 16px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 36px;
  color: rgba(0, 0, 0, 0.9);
}

.Billing,
.Shipping,
.Summary {
  width: 100%;
}

.Sidebar {
  transition: all 0.6s;
  padding: 16px 32px;
}

.IndexGuest a {
  color: #0a75c2;
  text-decoration: none;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}


.IndexGuest .stx-select__select-element,
.IndexGuest .stx-input {
  border-radius: 4px;
}

.IndexGuest a:visited {
  color: #0a75c2;
}

.IndexGuest__Customer,
.IndexGuest__ShippingAddress,
.IndexGuest__ShippingMethod,
.IndexGuest__Payment{
  padding: 16px 0;
}

.IndexGuest__Summary {
  padding: 16px 0;
  border-width: 1px 0;
  border-style: solid;
  border-color: #D9D9D9;
}

.IndexGuest__Title,
.IndexGuest__Summary__Btn h2,
.IndexGuest__Summary .stx-price__amount,
.IndexGuest__ShippingMethod .FieldSet__Heading {
  font-size: 18px;
  line-height: 24px;
  color: rgba(0, 0, 0, 0.9);
  font-weight: bold;
  margin: 0;
}

.IndexGuest__Summary__Btn {
  display: flex;
  border: none;
  background-color: unset;
  width: 100%;
  align-items: center;
  cursor: pointer;
}

.IndexGuest__Summary__Btn h2 {
  flex-grow: 1;
  text-align: left;
}

.IndexGuest__Title,
.IndexGuest__ShippingMethod .FieldSet__Header {
  margin-bottom: 14px;
}

.IndexGuest__Customer {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
}

.IndexGuest__Customer .stx-field--with-input {
  margin: 8px 0;
}

.IndexGuest__Customer .stx-input {
  height: 2.25rem;
  padding: 0 0.5rem;
}

.IndexGuest__Customer .stx-checkbox-field__checkbox {
  height: 16px;
  width: 16px;
}
.IndexGuest__Customer .stx-checkbox-field__field {
  padding-left: 26px;
}

.Billing-CheckoutButton:disabled {
  background-color: rgba(0, 0, 0, 0.4);
}

@media only screen and (max-width: 768px) {
  .Sidebar {
    height: 100%;
    overflow-y: auto;
  }

  .Billing,
  .Shipping,
  .Summary {
    position: absolute;
    left: 0;
    top: 0;
  }

  .Sidebar--Show {
    transform: translate(0, 0);
    visibility: visible;
  }

  .Sidebar--Hide {
    transform: translate(100%, 0);
    visibility: hidden;
  }

  .IndexPage--Hide {
    transform: translate(-100%, 0);
    visibility: hidden;
  }
}

@media only screen and (min-width: 769px) {
  .Checkout__Main {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .Checkout__Form {
    transition: all 0.6s;
    overflow-y: hidden;
    max-height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  .IndexPage--Hide {
    opacity: 0;
    display: none;
  }

  .IndexPage--Show {
    opacity: 1;
    display: block;
  }

  .Sidebar--Hide {
    opacity: 0;
    visibility: hidden;
    display: none;
  }

  .Sidebar--Show {
    opacity: 1;
    visibility: visible;
    display: block;
  }
}