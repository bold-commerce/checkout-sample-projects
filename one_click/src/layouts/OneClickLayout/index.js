export { default as OneClickLayout } from './OneClickLayout';