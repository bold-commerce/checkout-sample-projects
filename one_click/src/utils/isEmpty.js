const isEmpty = (obj) => {
  return JSON.stringify(obj) === JSON.stringify({});
};

export default isEmpty;