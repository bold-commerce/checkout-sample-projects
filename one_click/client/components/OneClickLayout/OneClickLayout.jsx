import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { MemoryRouter as Router } from "react-router-dom";
import { useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { CheckoutForm } from './components/CheckoutForm';
import { LayoutContext } from './context/LayoutContext';
import { Message} from '@boldcommerce/stacks-ui';
import './OneClickLayout.css';

const OneClickLayout = ({ orderStatus, orderErrors }) => {
  const [openModal, setOpenModal] = useState(true);

  return (
    <LayoutContext.Provider value={{openModal, setOpenModal}}>
      <div className="Checkout__Main">
        { orderErrors['order'] &&
          <Message type="alert">
            An error with your order has occured, please try again
          </Message>
        }
        <Router>
          <CheckoutForm />
        </Router>
      </div>
    </LayoutContext.Provider>
  );
};

OneClickLayout.propTypes = {
  orderStatus: PropTypes.string,
};

const MemoizedOneClickLayout = React.memo(OneClickLayout);

const OneClickContainer = () => {
  const { state } = useCheckoutStore();
  const { orderStatus } = state.orderInfo;
  const orderErrors  = state.errors;

  return <MemoizedOneClickLayout orderState={orderStatus} orderErrors={orderErrors} />;
};

export default OneClickContainer;
