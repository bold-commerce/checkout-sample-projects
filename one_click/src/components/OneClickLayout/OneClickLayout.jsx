import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { MemoryRouter as Router } from "react-router-dom";
import { useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { CheckoutForm } from './components/CheckoutForm';
import { LayoutContext } from './context/LayoutContext';
import { Message} from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';
import './OneClickLayout.css';

const OneClickLayout = ({ orderErrors }) => {
  const { t } = useTranslation(); 
  const [openModal, setOpenModal] = useState(true);

  return (
    <LayoutContext.Provider value={{openModal, setOpenModal}}>
      <div className="Checkout__Main">
        {
          orderErrors['order'] &&
          <Message type="alert">
            {t('error.order')}
          </Message>
        }
        <Router>
          <CheckoutForm />
        </Router>              
      </div>
    </LayoutContext.Provider>
  );
};

OneClickLayout.propTypes = {
  orderStatus: PropTypes.string,
};

const MemoizedOneClickLayout = React.memo(OneClickLayout);

const OneClickContainer = () => {
  const { state } = useCheckoutStore();
  const { orderStatus } = state.orderInfo;
  const orderErrors  = state.errors;

  return <MemoizedOneClickLayout orderState={orderStatus} orderErrors={orderErrors} />;
};

export default OneClickContainer;
