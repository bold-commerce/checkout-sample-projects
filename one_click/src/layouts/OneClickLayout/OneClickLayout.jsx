import React, {  } from 'react';
import PropTypes from 'prop-types';
import { MemoryRouter as Router } from "react-router-dom";
import { useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { CheckoutForm } from '../../components/CheckoutForm';
import { Message} from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';
import './OneClickLayout.css';

const OneClickLayout = ({ orderErrors }) => {
  const { t } = useTranslation();
  const orderError = Boolean(orderErrors['order']);
  return (
    <div className="Checkout__Main">
      {
        orderError &&
        <Message type="alert">
          {t('error.order')}
        </Message>
      }
      <Router>
        <CheckoutForm banners={orderError} />
      </Router>              
    </div>
  );
};

OneClickLayout.propTypes = {
  orderStatus: PropTypes.string,
};

const MemoizedOneClickLayout = React.memo(OneClickLayout);

const OneClickContainer = () => {
  const { state } = useCheckoutStore();
  const { orderStatus } = state.orderInfo;
  const orderErrors  = state.errors;

  return <MemoizedOneClickLayout orderState={orderStatus} orderErrors={orderErrors} />;
};

export default OneClickContainer;
