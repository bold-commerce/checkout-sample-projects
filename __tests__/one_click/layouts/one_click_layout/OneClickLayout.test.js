import React from "react";
import { render, waitFor } from '@testing-library/react';
import { OneClickLayout } from '../../../../one_click/src/layouts/OneClickLayout'
import ResizeObserver from "../../../../__mocks__/ResizeObserver";
import { exampleLineItems as MOCKexampleLineItems } from '../../../utils/lineItemHelpers';
import { testApplicationState as MOCKexampleApplicationState } from '../../../utils/applicationStateHelper';
import {
    countries as MOCKcountries,
    caProvinces as MOCKprovinces,
    exampleAddress as MOCKexampleAddress,
    exampleSavedAddresses as MOCKexampleSavedAddresses
} from '../../../utils/addressHelpers';
import { exampleShippingState as MOCKexampleShippingState } from '../../../utils/shippingLinesHelper';
import '../../../../one_click/src/i18n/config'

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useLineItems: () => ({
        data: MOCKexampleLineItems,
        updateLineItemQuantity: (() => {}),
        removeLineItem: (() => {})        
    }),
    useCheckoutStore: () => ({
        state: {
            orderInfo: {
                billingSameAsShipping:true,
                orderStatus: ""
            },
            errors: { order: [] },
            loadingStatus: { isLoading: false },
            applicationState: MOCKexampleApplicationState,
            orderTotals: {
                taxesTotal: 1200,
                subTotal: 23000,
                total: 24200
            }
        }
    }),
    useCustomer: () => ({
        data: {
            platform_id: 123
        }
    }),
    usePaymentIframe: () => ({
        data: {
            url: 'test.url',
            loadingStatus: 'fulfilled',
            paymentIframeOnLoaded: (() => {})
        }
    }),
    useSavedAddresses: () => ({
        data: MOCKexampleSavedAddresses
    }),
    useShippingLines: () => ({
        data: MOCKexampleShippingState
    }),
    useShippingAddress: () => ({
        data: MOCKexampleAddress,
        submitShippingAddress: (() => {})
    }),
    useBillingAddress: () => ({
        data: MOCKexampleAddress
    }),
    useDiscount: () => ({
        data: '?',
        applyDiscount: (() => {})
    }),
    useBillingSameAsShipping: () => ({
        data: true,
        setBillingSameAsShipping: (() => {})
    }),
    useCountryInfo: () => ({
        data: {
            countries: MOCKcountries,
            provinceLabel: "province",
            provinces: MOCKprovinces,
            showPostalCode: true,
            showProvince: true
        }
    }),
    useLoadingStatus: () => ({
        shippingAddressLoadingStatus: 'complete',
        shippingLinesLoadingStatus: 'complete'
    })
})).mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        websiteName: 'TestSite'
    })
})).mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useLocation: () => ({
        pathname: '/' 
    }),
    useNavigate: (() => {})
})).mock('../../../../one_click/src/hooks', () => ({
    ...jest.requireActual('../../../../one_click/src/hooks'),
    useInventory: () => ({
        checkInventory: (() => {})
    })
}));

describe('OneClickLayout', () => {
    test('renders OneClickLayout component', async () => {
        const { asFragment } = render( <OneClickLayout /> );
        await waitFor(() => {
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
