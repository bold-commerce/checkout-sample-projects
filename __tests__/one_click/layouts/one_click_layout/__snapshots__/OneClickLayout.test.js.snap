// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneClickLayout renders OneClickLayout component 1`] = `
<DocumentFragment>
  <div
    class="Checkout__Main"
  >
 
  </div>
</DocumentFragment>
`;
